File: Old_items/Toph/Criterion10/C.py
mod = 1000000007
a = []
a.append(1)
for i in range(1, 20000002):
	a.append((a[i - 1] * i) % mod)
for _ in range(int(input())):
	n = int(input())
	print((a[n - 1] * a[2 * n]) % mod)

File: Old_items/Toph/Criterion10/C.cpp
#include<bits/stdc++.h>

#define ll long long
#define sc(n) scanf("%d", &n)
using namespace std;

const ll N = 2e7 + 3, mod = 1e9 + 7;
ll ans[N];

int main()
{
  ans[0] = 1;
  for (ll i = 1; i < N; i++)
    ans[i] = (ans[i - 1] * i) % mod;

  int tt, n;
  sc(tt);

  while (tt--)
  {
    sc(n);
    printf("%lld\n", (ans[n * 2] * ans[n - 1]) % mod);
  }
  return 0;
}